crates/webidl/src/idl_type.rs

```diff
@@ -88,6 +88,110 @@ pub(crate) enum IdlType<'a> {
     UnknownInterface(&'a str),
 }
 
+/// Indication that the [syn::Type] returned along with this does not tell the whole truth encoded
+/// in the IDL. All its properties are shaped by which information it needs in its `.tell()` method
+/// that converts the gathered information into something that can be put into the docs.
+#[derive(Debug)]
+pub struct Confession {
+    kind: ConfessionKind,
+    actual_type: Option<String>,
+    is_behind_result: bool,
+    has_more_data: bool,
+}
+
+#[derive(Debug)]
+enum ConfessionKind {
+    Promise,
+    Iterable,
+}
+
+impl Confession {
+    /// Create a textual description of what's wrong with the returned type.
+    ///
+    /// Any formatting in the return value is rustdoc-suitable Markdown.
+    pub fn tell(&self, in_return_position: bool) -> String {
+        (|| -> Result<_, std::fmt::Error> {
+            use std::fmt::Write;
+            let mut text = String::new();
+            let mut show_more = self.has_more_data;
+
+            match (&self.kind, &self.actual_type) {
+                (ConfessionKind::Promise, Some(s)) => {
+                    write!(text, "While the Promise can produce any JsValue as far as the type system is \
+                            concerned, practically it is expected to contain a `{}`.", s)?;
+                    if in_return_position {
+                        write!(text, " It can be converted like \
+                               `let result: {} = {}.await.into();`.",
+                            s,
+                            if self.is_behind_result { "result?" } else { "result" }
+                            )?;
+                    }
+                },
+                (ConfessionKind::Promise, None) => {
+                    show_more = false;
+                    write!(text, "There is additional information in the IDL file about the content of \
+                           the promise, but it can not yet be explained any better.")?;
+                }
+                (ConfessionKind::Iterable, Some(s)) => {
+                    write!(text, "While the iterable or array can produce any JsValue as far as the \
+                            type system is concerned, practically it is expected to contain a `{}`.", s)?;
+                },
+                (ConfessionKind::Iterable, None) => {
+                    show_more = false;
+                    write!(text, "There is additional information in the IDL file about the items in \
+                           this iterable or array, but it can not yet be explained any better.")?;
+                }
+            }
+
+            if show_more {
+                write!(text, " More information is available in the source IDL file.")?;
```

Comment:
Why not change that? I'd think the root of any generated crate is a good point to state what all these items were generated from; eg. like in 395256169eb6c827e8236fdc4f9aa441c8180753. (If that stays, maybe I can make that statement linkable from the confessions, but not sure how well that works given it can be used with different -sys crates potentially).

Reply:
I think unless this links to the WebIDL file in question it won't be too useful. Otherwise it's just saying "more info is probably somewhere" but it's not clear where to go to get more info.